doc/source/operations/nova-compute-ironic.rst

```diff
@@ -0,0 +1,211 @@
+
+  .. code-block:: console
+
+    $ ssh controller1
+    $ sudo docker exec -it mariadb mysql -u nova -p$(sudo grep 'mysql+pymysql://nova:' /etc/kolla/nova-api/nova.conf | awk -F'[:,@]' '{print $3}')
+    $ MariaDB [(none)]> use nova;
+
+  Proceed with caution. It is good practise to update one record first:
+
+  .. code-block:: console
+
+    $ MariaDB [nova]> update instances set host='controller1-ironic' where uuid=0 and host='controller3-ironic' limit 1;
+      Query OK, 1 row affected (0.002 sec)
+      Rows matched: 1  Changed: 1  Warnings: 0
+
+  At this stage you should go back to step 4 and check that the numbers have
+  changed as expected. When you are happy, update remaining records for all
+  services which have been removed:
+
+  .. code-block:: console
+
+    $ MariaDB [nova]> update instances set host='controller1-ironic' where deleted=0 and host='controller3-ironic';
+      Query OK, 59 rows affected (0.009 sec)
+      Rows matched: 59  Changed: 59  Warnings: 0
+    $ MariaDB [nova]> update instances set host='controller1-ironic' where deleted=0 and host='controller2-ironic';
+      Query OK, 35 rows affected (0.003 sec)
+      Rows matched: 35  Changed: 35  Warnings: 0
+
+9. Repeat step 4. Verify that all Ironic nodes are using the single remaining
+   Nova Compute Ironic instance.
+
+
+Making it easy to re-deploy Nova Compute Ironic
+-----------------------------------------------
+
+In the previous section we saw that at any given time, a baremetal node is
+associated with a single Nova Compute Ironic instance. At this stage, assuming
+that you have diligently followed the instructions, you are in the situation
+where all Ironic baremetal nodes are managed by a single Nova Compute Ironic
+instance. If this service goes down, you will not be able to manage /any/
+baremetal nodes.
+
+By default, the single remaining Nova Compute Ironic instance will be named
+after the host on which it is deployed. The host name is passed to the Nova
+Compute Ironic instance via the default section of the ``nova.conf`` file,
+using the field: ``host``.
+
+If you wish to re-deploy this instance, for example because the original host
+was permanently mangled in the World Server Throwing Championship [2], you
+must ensure that the new instance has the same name as the old one. Simply
+setting ``kolla_nova_compute_ironic_host`` to another controller and
+re-deploying the service is not enough; the new instance will be named after
+the new host.
+
+To work around this you should set the ``host`` field in ``nova.conf`` to a
+constant, such that the new Nova Compute Ironic instance comes up with the
+same name as the one it replaces.
+
+For example, if the original instance resides on ``controller1``, then set the
+following in ``etc/kayobe/nova.yml``:
+
+.. code-block:: console
+
+  kolla_nova_compute_ironic_static_host_name: controller1-ironic
+
+Note that an ``-ironic`` postfix is added to the hostname. This comes from
```

Comment:
I'm adding some notes on running the Kayobe commands to re-deploy the service. And considerations that need to be made if the old host is offline etc. (breaking the service deploy, pending 'Kayobe reliability' changes, etc.